### PR TITLE
Audio Effects Cleanup

### PR DESCRIPTION
**Pull request by engineer124**
Thus begins the slow untangling of `z64audio.h`.

It's be nice if there was a way to not have to include `audio/effects.h` inside `z64audio.h`, but I think I have to for now

### CODE REVIEW

File: include/audio/effects.h [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+#include "PR/ultratypes.h"
+#include "unk.h"
+
+struct Note;
+struct SequencePlayer;
+
+/* Multi-Point ADSR */
+
+typedef enum AdsrStatus {
```
Comment (AngheloAlf) [inline include/audio/effects.h:12]: What's an `ADSR`?

File: src/audio/lib/effects.c [hunk 2]
```diff
@@ -1,14 +1,26 @@
...
+void AudioScript_SequenceChannelProcessSound(SequenceChannel* channel, s32 recalculateVolume, s32 applyBend) {
     f32 channelVolume;
     f32 chanFreqScale;
     s32 i;
 
     if (channel->changes.s.volume || recalculateVolume) {
         channelVolume = channel->volume * channel->volumeScale * channel->seqPlayer->appliedFadeVolume;
         if (channel->seqPlayer->muted && (channel->muteFlags & MUTE_FLAGS_SOFTEN)) {
-            channelVolume = channel->seqPlayer->muteVolumeScale * channelVolume;
+            channelVolume = channelVolume * channel->seqPlayer->muteVolumeScale;
```
Comment (AngheloAlf) [inline src/audio/lib/effects.c:23]: `*=` doesn't work, right?

**Review by AngheloAlf — APPROVED** [review]

File: include/audio/effects.h (hunk 1 above)
Comment (engineer124) [reply include/audio/effects.h:12]: It stands for **Attack, Decay, Sustain, Release**, it describes how you play a note's volume over the entire duration of a note:
https://en.wikipedia.org/wiki/Envelope_(music)
https://www.masterclass.com/articles/adsr-envelope-explained

File: src/audio/lib/effects.c (hunk 2 above)
Comment (engineer124) [reply src/audio/lib/effects.c:23]: Unfortunately not, swaps the floating point registers

File: include/audio/effects.h (hunk 1 above)
Comment (engineer124) [reply include/audio/effects.h:12]: I'd also like to add for any audio enthusiasts who may stumble on this PR, a todo outside the scope of this PR is to address that ADSR might not be the best name for this process. SM64/OoT/MM/AF all treat this volume envelope as more of a script with commands that can make a custom shape (to what extent I haven't explored fully) but it's not really limited to the ADSR shape you commonly see:

image

Probably the best thing is to just stick with `ADSR` -> `Envelope` in all cases, and I think this was starting to be address in the audio extraction PR. But I'd like to explore Envelopes more before making that change. Some of the names like `delay` and `release` in master for adsr may need tweaking too. Soon :tm:

File: include/audio/effects.h [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    /* 0 */ ADSR_STATE_DISABLED,
+    /* 1 */ ADSR_STATE_INITIAL,
+    /* 2 */ ADSR_STATE_START_LOOP,
+    /* 3 */ ADSR_STATE_LOOP,
+    /* 4 */ ADSR_STATE_FADE,
+    /* 5 */ ADSR_STATE_HANG,
+    /* 6 */ ADSR_STATE_DECAY,
+    /* 7 */ ADSR_STATE_RELEASE,
+    /* 8 */ ADSR_STATE_SUSTAIN
+} AdsrStatus;
```
Comment (hensldm) [inline include/audio/effects.h:22]: Are these meant to be `state`s or `status`es? Also want to be careful with conflicting with the `AdsrState` struct (which is probably why the enum name is `Status`.

File: src/audio/lib/effects.c [hunk 4]
```diff
@@ -104,13 +119,17 @@ s16 AudioEffects_GetVibratoPitchChange(VibratoState* vib) {
     s32 index;
 
     vib->time += (s32)vib->rate;
-    index = (vib->time >> 10) & 0x3F;
+    // 0x400 is 1 unit of time, 0x10000 is 1 period
+    index = (vib->time >> 10) % WAVE_SAMPLE_COUNT;
```
Comment (hensldm) [inline src/audio/lib/effects.c:123]:
```suggestion
    index = (vib->time / 0x400) % WAVE_SAMPLE_COUNT;
```
? 
Seems to work, and would help connect it to the comment

File: include/audio/effects.h (hunk 3 above)
Comment (engineer124) [reply include/audio/effects.h:22]: Oh nice catch. Yeah I'll review and see which is appropriate where

**Review by hensldm — APPROVED** [review]

**Review by jpburnett — APPROVED** [review]
pepeJAM

**Pull request merged**